hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -979,6 +1066,9 @@ public S3AFileStatus getFileStatus(Path f) throws IOException {
 
         return new S3AFileStatus(true, false,
             f.makeQualified(uri, workingDir));
+      } else if (key.isEmpty()) {
+        LOG.debug("Found root directory");
```

Comment:
This is a change I do not see in the open-source commit. Could you please explain why we need this?